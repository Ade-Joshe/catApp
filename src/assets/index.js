export * from "./allCatIcon";
export * from "./catsILikeIcon";
export * from "./emptyHeart";
export * from "./filledHeart";