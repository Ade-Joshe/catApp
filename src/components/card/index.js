export * from "./miniCard";
export * from "./emptyCard";
export * from "./largeCard";