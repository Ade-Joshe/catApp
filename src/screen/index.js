export * from "./favouriteScreen";
export * from "./homeScreen";