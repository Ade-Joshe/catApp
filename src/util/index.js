export * from "./pixelConverter";
export * from "./theme";
export * from "./store";