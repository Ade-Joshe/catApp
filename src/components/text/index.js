import React, { useContext } from 'react'
import { StyleSheet, Text } from 'react-native'
import { fontPixel, heightPixel, ThemeContext } from '../../util'

export const PageTitle = ({ children }) => {
    const { colors } = useContext(ThemeContext);
    return <Text style={[styles.title, { color: colors.black, }]}>{children}</Text>
};

const styles = StyleSheet.create({
    title: { fontSize: fontPixel(16), marginVertical: heightPixel(25), fontWeight: "600" }
})
