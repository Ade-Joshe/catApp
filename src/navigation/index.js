import React, { useContext } from "react"
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { NavigationContainer } from "@react-navigation/native";
import { HomeScreen, FavouriteScreen } from '../screen';
import { AllCatIcon, CatsILikeIcon } from "../assets";
import { fontPixel, ThemeContext } from "../util";

const Tab = createBottomTabNavigator();

export function RootNavigation() {

    const { colors } = useContext(ThemeContext);

    return (
        <NavigationContainer>
            <Tab.Navigator
                screenOptions={({ route }) => ({
                    unmountOnBlur: true,
                    headerShown: false,
                    lazy: true,
                    tabBarActiveTintColor: colors.black,
                    tabBarInactiveTintColor: colors.grey,
                    tabBarStyle: { height: 67 },
                    tabBarLabelStyle: { fontSize: fontPixel(15) },
                    tabBarLabelPosition: 'below-icon',
                    tabBarIcon: ({ focused }) => {
                        if (route.name === 'All Cats') return <AllCatIcon color={focused ? colors.black : colors.grey} />;
                        else if (route.name === 'Cats I Like') return <CatsILikeIcon color={focused ? colors.black : colors.grey} />;
                        else return;
                    }
                })}
            >
                <Tab.Screen name="All Cats" component={HomeScreen} />
                <Tab.Screen name="Cats I Like" component={FavouriteScreen} />
            </Tab.Navigator>
        </NavigationContainer >
    );
}