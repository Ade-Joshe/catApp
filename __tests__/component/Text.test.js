import React from "react"
import { render } from '@testing-library/react-native';
import { PageTitle } from "../../src/components";
import { create } from "react-test-renderer";

it("renders correctly when children is passed", () => {
    const { getAllByText } = render(<PageTitle>Titles</PageTitle>);

    expect(getAllByText("Titles")).toBeTruthy();
});

it("snapshot testing for matching", () => {
    let component = create(<PageTitle />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});