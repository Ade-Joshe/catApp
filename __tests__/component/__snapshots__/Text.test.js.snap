// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing for matching 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 32,
        "fontWeight": "600",
        "marginVertical": 41,
      },
      Object {
        "color": "#212227",
      },
    ]
  }
/>
`;
